src/Datadog.NativeLoader/dllmain.cpp

```diff
@@ -0,0 +1,176 @@
+// dllmain.cpp : Defines the entry point for the DLL application.
+#include "class_factory.h"
+
+#include <filesystem>
+#include <fstream>
+#include <unordered_map>
+
+#include "logging.h"
+#include "pal.h"
+#include "proxy.h"
+
+using namespace datadog::nativeloader;
+
+const std::string conf_filename = "loader.conf";
+#if _WIN32
+const std::string dynExtension = ".dll";
+#elif LINUX
+const std::string dynExtension = ".so";
+#elif MACOS
+const std::string dynExtension = ".dylib";
+#endif
+
+DynamicDispatcher* dispatcher;
+
+// Gets the profiler path
+static WSTRING GetProfilerPath()
+{
+    WSTRING profiler_path;
+    profiler_path = GetEnvironmentValue(WStr("CORECLR_PROFILER_PATH"));
+    if (profiler_path.length() == 0)
+    {
+        profiler_path = GetEnvironmentValue(WStr("COR_PROFILER_PATH"));
+    }
+
+#if BIT64
+    if (profiler_path.length() == 0)
+    {
+        profiler_path = GetEnvironmentValue(WStr("CORECLR_PROFILER_PATH_64"));
+    }
+    if (profiler_path.length() == 0)
+    {
+        profiler_path = GetEnvironmentValue(WStr("COR_PROFILER_PATH_64"));
+    }
+#else
+    if (profiler_path.length() == 0)
+    {
+        profiler_path = GetEnvironmentValue(WStr("CORECLR_PROFILER_PATH_32"));
+    }
+    if (profiler_path.length() == 0)
+    {
+        profiler_path = GetEnvironmentValue(WStr("COR_PROFILER_PATH_32"));
+    }
+#endif
+
+    return profiler_path;
+}
+
+extern "C"
+{
+    BOOL STDMETHODCALLTYPE DllMain(HMODULE hModule, DWORD ul_reason_for_call, LPVOID lpReserved)
+    {
+        // Perform actions based on the reason for calling.
+        switch (ul_reason_for_call)
+        {
+            case DLL_PROCESS_ATTACH:
+            {
+                // Initialize once for each new process.
+                // Return FALSE to fail DLL load.
+
+                Debug("DllMain - DLL_PROCESS_ATTACH");
+
+                dispatcher = new DynamicDispatcher();
```

Comment:
ok if I understand it correctly, the profiler and the tracer libraries will be at the same level as the proxy? I forgot, but I'm wondering if in the doc Greg was not talking about having their own subdir.